File: 07_study/Gyeongwook/findPeakElement.py
class Solution:
    # @param nums, an integer[]
    # @return an integer
    def findPeakElement(self, nums):
        elementCount = len(nums)
        
        startIdx = 0
        endIdx = elementCount - 1
        mid = endIdx / 2
        
        while startIdx < endIdx :
            if (mid == startIdx or nums[mid-1] < nums[mid]) and (mid == endIdx or nums[mid] > nums[mid+1]) :
                break
            else :
                if num > startIdx and nums[mid-1] > nums[mid] :
                    endIdx = mid - 1
                elif mid < endIdx and nums[mid] < nums[mid+1] :
                    startIdx = mid + 1
        			
            mid = endIdx + (startIdx - endIdx) / 2
            
        return mid

File: 07_study/Seonghun Kim/findPeakElement.c
int findPeakElement(int* nums, int numsSize) {
        int dir = 1;
        for (int i = 1 ; i < numsSize ; i++) {
            int newDir = nums[i] - nums[i-1];
            if (newDir * dir < 0)
                return i-1;
            dir = newDir;
        }
        
        if (dir * -1 < 0)
            return numsSize-1;    
            
        return 0;    
}